### DISCUSSION TITLE
Documentation on $.breezemap.__register and JS Component.register

### DISCUSSION THREAD

**Discussion by sandipansaha** [breezefront/community#297] [2. Ideas]
Hi,

I am not sure whether we have this documented, however, I would love to know more about their usage. These are being used in the enterprise integration module. If possible we can a area in documentation.

- $.breezemap.__register 
- Component.register

```php
    $.breezemap.__register('Magento_GiftWrapping/js/model/gift-wrapping', 0);
    $.breezemap.__register('Magento_GiftWrapping/js/model/gift-wrapping-collection', 1);
``` 
```php
Component.register('Magento_PageBuilderProductRecommendations/js/pageBuilderRenderer');
``` 
Thanks

**Comment by vovayatsyuk** [comment]
This code is not described in the docs yet because I wouldn't say I like it :)

What this code does, is allow reusing Luma's components without modifications - without adding `component` property.

**Here is how it works.**

When Breeze processes the code like this:

```js
define([], () => {
    return ... // function, or object, or uiComponent
})
```

We register the returned value in $.breezemap as `__component{counter}`.

Now, we need to register that component as `Magento_GiftWrapping/js/model/gift-wrapping` or whatever it is. To do so, we include and execute our file right after the original file from Luma:

```js
$.breezemap.__register('Magento_GiftWrapping/js/model/gift-wrapping'); // <-- same as Component.register
```

p.s. To make it all work, we must guarantee that all deps of `define([...])` were resolved before code execution. Otherwise `$.breezemap.__register` will be executed before we register the anonymous component. That's why `imports` in XML files are important now.
